qulice-checkstyle/src/main/java/com/qulice/checkstyle/LineRanges.java

```diff
@@ -0,0 +1,153 @@
+ * products derived from this software without specific prior written
+ * permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT
+ * NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.qulice.checkstyle;
+
+import com.google.common.base.Predicate;
+import com.google.common.collect.FluentIterable;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+
+/**
+ * Represent a set of LineRange objects. For example, an instance of this class
+ * could represent all the line ranges for methods in a given Java source code
+ * file.
+ *
+ * @author Jimmy Spivey (JimDeanSpivey@gmail.com)
+ * @version $Id$
+ */
+public final class LineRanges {
+
+    /**
+     * ArrayList of line ranges.
+     */
+    private final transient LineRanges.LocalCollection lines =
+        new LineRanges.LocalCollection();
+
+    /**
+     * Adds a line range to the internal collection.
+     * @param line The line range to add to the collection
+     */
+    public void add(final LineRange line) {
+        this.lines.collection().add(line);
+    }
+
+    /**
+     * Expses the iterator of the internal collection.
+     * @return Iterator pointing to the internal collections elements.
+     */
+    public Iterator<LineRange> iterate() {
+        return this.lines.collection().iterator();
+    }
+
+    /**
+     * Detects if the proposed line number is with any of the line ranges.
+     * @param line The proposed line number to check
+     * @return True if the proposed line number is within any line range.
+     */
+    public boolean inRange(final int line) {
+        return !this.lines.collection().isEmpty()
+            && FluentIterable.from(this.lines.collection())
+            .anyMatch(new LineRanges.LineWithAny(line));
+    }
+
+    /**
+     * Gets the subset of LineRanges that are within all proposed ranges. Does
+     * not return null; instead, returns empty range if there are no matches.
+     * @param ranges The ranges to filter on.
+     * @return Returns all LineRange elements that are within range.
+     */
+    public LineRanges within(final LineRanges ranges) {
+        final LineRanges result = new LineRanges();
+        final Iterator<LineRange> iterator = ranges.iterate();
+        while (iterator.hasNext()) {
```

Comment:
@pinaf I would have to break encapsulation and expose the internal reference to the collection since `foreach` can only iterate over an `Iterable` and not an `Iterator`.